main.go

```diff
@@ -210,16 +221,47 @@ func main() {
 			Port:    webhookPort,
 			TLSOpts: tlsOptionOverrides,
 		}),
-		LeaderElection:          enableLeaderElection,
-		LeaderElectionID:        leaderElectionID,
-		LeaderElectionNamespace: leaderElectionNamespace,
-		HealthProbeBindAddress:  healthAddr,
+		LeaderElection:                enableLeaderElection,
+		LeaderElectionID:              leaderElectionID,
+		LeaderElectionNamespace:       leaderElectionNamespace,
+		LeaderElectionReleaseOnCancel: leaderElectionReleaseOnCancel,
+		HealthProbeBindAddress:        healthAddr,
 		Cache: cache.Options{
 			ByObject:          secretutils.AddSecretSelector(nil),
 			DefaultNamespaces: watchNamespaces,
 		},
 	}
 
+	if leaseDuration != "" {
+		dur, err := time.ParseDuration(leaseDuration)
```

Comment:
Looking at the defaults, it's tempting to skip the parsing and make the command line argument `-lease-duration-seconds`. Nobody is going to want a duration in years or even hours. It could be a float if they really want to get down to the milliseconds.

Reply:
If we have `leaseDurationSeconds`, it'll create weird temporaries like:

```go
leaseDurationSecondsDuration := time.Duration(leaseDurationSeconds) * time.Second
ctrlOptions.LeaseDuration = &leaseDurationSecondsDuration
```

But more annoyingly, we'll have to parse the env vars to `int`s. So this tidy line:

```go
flag.StringVar(&renewDeadline, "renew-deadline", os.Getenv("RENEW_DEADLINE"), "Golang duration, e.g. '26s'")
```

would become something more involved because `os.Getenv` returns a string but the default must be a number.

Reply:
> it'll create weird temporaries

TBH I would take that one extra line of code over complicated user documentation.

> we'll have to parse the env vars to ints

`flag.IntVar()` is only a convenience. If you wanted to keep it as `StringVar` and continue to do the parsing to int/float here yourself I don't think there's anything stopping you.

Reply:
I switched it to using seconds, left it as a `StringVar` and added `strconv.ParseInt`.